From 19764e461710f5af8fb8e073a63f9a5a8ae431af Mon Sep 17 00:00:00 2001
From: dvshkurk <Dmitry.V.Shkurko@intel.com>
Date: Tue, 5 Jun 2012 17:21:27 +0700
Subject: [PORT FROM MAIN] Do not use __builtin_unreachable with Intel compiler

BZ:45394

Change-Id: Ie71e7c9d85e252e1cdb5d99dde4a3c728992711e
Orig-Change-Id: Ia49e5255b2705fbd3fed79cf2769879d5f52b20f
Signed-off-by: dvshkurk <Dmitry.V.Shkurko@intel.com>
Reviewed-on: http://android.intel.com:8080/57688
Reviewed-by: Chen, Yong A <yong.a.chen@intel.com>
Reviewed-by: Shkurko, Dmitry V <dmitry.v.shkurko@intel.com>
Reviewed-by: Yao, Yong <yong.yao@intel.com>
Reviewed-by: Beare, Bruce J <bruce.j.beare@intel.com>
Reviewed-by: Li, XiaojingX <xiaojingx.li@intel.com>
Tested-by: Li, XiaojingX <xiaojingx.li@intel.com>
Reviewed-by: buildbot <buildbot@intel.com>
Tested-by: buildbot <buildbot@intel.com>
Reviewed-on: http://android.intel.com:8080/78279
Reviewed-by: cactus <cactus@intel.com>
Tested-by: cactus <cactus@intel.com>
---
 include/llvm/Support/Compiler.h |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/include/llvm/Support/Compiler.h b/include/llvm/Support/Compiler.h
index 25f42a9..d52a9b2 100644
--- a/include/llvm/Support/Compiler.h
+++ b/include/llvm/Support/Compiler.h
@@ -268,7 +268,7 @@
 /// to an expression which states that it is undefined behavior for the
 /// compiler to reach this point.  Otherwise is not defined.
 #if defined(__clang__) || (__GNUC__ > 4) \
- || (__GNUC__ == 4 && __GNUC_MINOR__ >= 5)
+ || (__GNUC__ == 4 && __GNUC_MINOR__ >= 5) && !defined(__INTEL_COMPILER)
 # define LLVM_BUILTIN_UNREACHABLE __builtin_unreachable()
 #elif defined(_MSC_VER)
 # define LLVM_BUILTIN_UNREACHABLE __assume(false)
-- 
1.7.4.1

